src/AggregateFunctions/AggregateFunctionMap.h

```diff
@@ -84,6 +84,31 @@ class AggregateFunctionMap final
     using Base = IAggregateFunctionDataHelper<Data, AggregateFunctionMap<KeyType>>;
 
 public:
+    bool isState() const override
+    {
+        return nested_func->isState();
+    }
+
+    bool isVersioned() const override
+    {
+        return nested_func->isVersioned();
+    }
+
+    size_t getVersionFromRevision(size_t revision) const override
+    {
+        return nested_func->getVersionFromRevision(revision);
+    }
+
+    size_t getDefaultVersion() const override
+    {
+        return nested_func->getDefaultVersion();
+    }
+
+    bool hasTrivialDestructor() const override
+    {
+        return false;
+    }
```

Comment:
Also why is it not trivial?

Reply:
hasTrivialDestructor() should return `true` if it's not necessary to delete this data (call destroy()). So, we should return `false` for state of `-Map` combinator because it contains `unordered_map` that should be destroyed correctly

Reply:
but it already must return false. In the base class it returns`std::is_trivially_destructible_v<Data>` which is `false`.

Reply:
Oh, yes, you are right, didn't see this in base class, thanks! WIll remove it